src/lib/ext-activitylog.js

```diff
@@ -51,6 +54,12 @@ class Model {
     return this.filter.type.has(logType);
   }
 
+  matchFilterApiName(name) {
+    return (
+      this.filter.name.has(name) || this.filter.contentScriptUrls.has(name)
```

Comment:
Don't save `contentScriptUrls` - it is not used elsewhere.

When a filter is not applicable, just return true. You should also do this in your timestamp patch - when there is no timestamp filter don't check the filter at all.

```
if (type is content script) {
  return true;
}
return match normal filter otherwise;
```